src/wp-includes/theme.php

```diff
@@ -838,7 +838,9 @@ function switch_theme( $stylesheet ) {
 
 	// Clear pattern caches.
 	$new_theme->delete_pattern_cache();
-	$old_theme->delete_pattern_cache();
+	if ( ! is_multisite() ) {
```

Comment:
I presume this is so pattern caches aren't deleted for any other site on the network who has this theme activated, right? If that's true, is there a reason we wouldn't want to do the same for the new theme above?

Reply:
+1, _any_ cache invalidation on switching themes should only happen on single sites, as we shouldn't affect other sites in a multisite network that may use one of these themes.
```suggestion
	if ( ! is_multisite() ) {
		$new_theme->delete_pattern_cache();
```